File: market/test/java/ru/yandex/market/grade/statica/web/TagControllerTest.java
package ru.yandex.market.grade.statica.web;

import org.junit.Ignore;
import org.junit.Test;
import org.skyscreamer.jsonassert.JSONAssert;
import org.skyscreamer.jsonassert.JSONCompareMode;
import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;

import ru.yandex.common.util.IOUtils;
import ru.yandex.market.grade.statica.PersStaticWebTest;
import ru.yandex.market.pers.test.http.HttpClientMockUtils;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
import static ru.yandex.market.grade.statica.mock.SaasRequestMatchers.withQueryParam;

/**
 * @author varvara
 * 19.08.2019
 */
public class TagControllerTest extends PersStaticWebTest {

    @Ignore
    @Test
    public void modelTags() throws Exception {
        long modelId = 122344;
        //given
        setSaasResponse(
            "/data/saasresponse/tags/model_count.json",
            HttpClientMockUtils.and(withQueryParam("p=0"), withQueryParam("numdoc=1"))
        );

        setSaasResponse(
            "/data/saasresponse/tags/tags_zero_page.json",
            HttpClientMockUtils.and(withQueryParam("p=0"), withQueryParam("numdoc=10"), withQueryParam("gta=s_tag"))
        );

        setSaasResponse(
            "/data/saasresponse/tags/tags_first_page.json",
            HttpClientMockUtils.and(withQueryParam("p=1"), withQueryParam("numdoc=10"), withQueryParam("gta=s_tag"))
        );

        setSaasResponse(
            "/data/saasresponse/tags/tags_second_page.json",
            HttpClientMockUtils.and(withQueryParam("p=2"), withQueryParam("numdoc=10"), withQueryParam("gta=s_tag"))
        );

        setSaasResponse(
            "/data/saasresponse/tags/tags_third_page.json",
            HttpClientMockUtils.and(withQueryParam("p=3"), withQueryParam("numdoc=10"), withQueryParam("gta=s_tag"))
        );

        //when
        MockHttpServletRequestBuilder requestBuilder = get("/api/tags/model/" + modelId);
        String response = mockMvc.perform(requestBuilder)
            .andDo(print())
            .andExpect(status().is2xxSuccessful())
            .andReturn().getResponse().getContentAsString();

        //then

        JSONAssert.assertEquals(
            IOUtils.readInputStream(
                getClass().getResourceAsStream("/data/model_tags.json")),
            response, JSONCompareMode.STRICT_ORDER);
    }

    @Ignore
    @Test
    public void modelTagsOnePage() throws Exception {
        long modelId = 122344;
        //given
        setSaasResponse(
            "/data/saasresponse/tags/model_count_one_page.json",
            HttpClientMockUtils.and(withQueryParam("p=0"), withQueryParam("numdoc=1"))
        );

        setSaasResponse(
            "/data/saasresponse/tags/tags_only_one_page.json",
            HttpClientMockUtils.and(withQueryParam("p=0"), withQueryParam("numdoc=10"), withQueryParam("gta=s_tag"))
        );


        //when
        MockHttpServletRequestBuilder requestBuilder = get("/api/tags/model/" + modelId);
        String response = mockMvc.perform(requestBuilder)
            .andDo(print())
            .andExpect(status().is2xxSuccessful())
            .andReturn().getResponse().getContentAsString();

        //then

        JSONAssert.assertEquals(
            IOUtils.readInputStream(
                getClass().getResourceAsStream("/data/model_tags_one_page.json")),
            response, JSONCompareMode.STRICT_ORDER);
    }

}

File: market/test/integration/java/ru/yandex/market/logistics/yard/controller/TagControllerTest.kt
package ru.yandex.market.logistics.yard.controller

import org.junit.jupiter.api.Test
import org.mockito.Mockito
import org.springframework.http.MediaType
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
import org.springframework.test.web.servlet.result.MockMvcResultHandlers
import org.springframework.test.web.servlet.result.MockMvcResultMatchers
import ru.yandex.market.ff.client.dto.RequestTypeStatusMapContainer
import ru.yandex.market.logistics.yard.base.AbstractSecurityMockedContextualTest
import ru.yandex.market.logistics.yard.util.FileContentUtils

class TagControllerTest: AbstractSecurityMockedContextualTest() {

    @Test
    fun getTags() {
        val getRequestsResponse = readFromJson(
            "classpath:fixtures/v2/controller/tagcontroller/ff-wf-response.json",
            RequestTypeStatusMapContainer::class.java
        )
        val expected = FileContentUtils.getFileContent(
            "classpath:fixtures/v2/controller/tagcontroller/expected.json",
        )
        Mockito.`when`(ffWfApiClient.requestTypeStatusMap)
            .thenReturn(getRequestsResponse)

        mockMvc.perform(
            MockMvcRequestBuilders.get("/tags")
                .param("parentGroupId","3")
                .param("parentId","0")
                .contentType(MediaType.APPLICATION_JSON)
        )
            .andDo(MockMvcResultHandlers.print())
            .andExpect(MockMvcResultMatchers.content().json(expected))
            .andExpect(MockMvcResultMatchers.status().is2xxSuccessful)
    }

}
